Doodlebug.cpp

```diff
@@ -31,241 +29,163 @@ Doodlebug::Doodlebug(int x_pos, int y_pos):Critter(x_pos, y_pos)
    this->moveSuccess = 0;
 }
 
-/**********************************************************************
-** Description: This function moves the Doodlebug on the board.
-**  This function takes a Critter*** argument that represents the
-**  board the Doodlebugs are on. The two integer arguments represent
-**  the dimensions of the board.
-**********************************************************************/
 
+/**************************************************************************************************
+** Description: void move(Critter*** &, int, int))		   
+**		          The move function for the Doodle will first check to see if there are any ants in 
+**              the adjacent cells starting in the NORTH direction and proceed to check in the 
+**              CLOCKWISE direction until it has found an ant or all the cardinal directions have
+**              been checked.  If an ant is found to be near, the Doodlebug will eat the Ant.
+**              Otherwise, the Doodlebug will attempt to move by choosing a random direction 
+**              (simulated by randomly calling a number between 0-3), which represents the cardinal
+**              directions (NORTH, EAST, SOUTH, and WEST).  Based on the random direction, the 
+**              Doodlebug will attempt to move to that location as long as the move will not go out 
+**              of bounds and the space is unoccupied.
+**************************************************************************************************/
 void Doodlebug::move(Critter*** &boardIn, int size_x, int size_y)
 {
    int random = randIntRange(0,3);
    bool antNear = false;
    bool inBounds = false;
    bool cellFree = false;
-
    int new_x, new_y;
-   int attempts = 0;
 
    //check if ant is in an adjacent cell
-   //switch (random) 
-   //{
-      while(attempts < 4)
+   for (int j = 0; j < 4; j++)
+   {
+      //Check that a previous adjacent cell does not contain an ant
+      if (!antNear)
       {
-          if(random == 0)
```

Comment:
do these changes affect the fixed versions?

Reply:
It shouldn't affect anything aside from making the function attempting to move once if there is no ants around instead of trying to cycle through and finding an open adjacent cell if the first random spot doesn't work out.